### DISCUSSION TITLE
Fast get k-hop subgraphs for all nodes

### DISCUSSION THREAD

**Discussion by xiachenrui** [pyg-team/pytorch_geometric#7106] [Q&A] [answered]
Hi all,

   I want to get k-hop subgraphs for all nodes of a graph (5,000 nodes). What is the fastest method do you think ?
  My quick idea is using `NodeLoader()` with a big batch size. Can we use `Ray` to accelerate this ?
   Any suggestion is helpful !

Thanks !

**Comment by rusty1s** [comment, marked as answer]
Yeah, I would be using `NeighborLoader` for this. You can accelerate by increasing the `num_workers` argument. `ray` sounds like overkill for this task.

**Reply by xiachenrui** [reply]
Thanks for your helpful reply!
